### PR TITLE
fix: input alignment

### PR DESCRIPTION
**Pull request by KelvinOm**
## Description
Fixed the alignment of the content inside the input

## Summary by CodeRabbit

- **New Features**
  - Introduced `prefix` and `suffix` props for better customization in TextInput components, replacing `startIcon` and `endIcon`.
  - Added examples for using menus as prefixes/suffixes and for type password functionality in TextInput stories.
  - Implemented loading indicator functionality with priority over icons in TextInput.
  - Added support for disabled and readonly states with icon prefixes and suffixes in TextInput.

- **Bug Fixes**
  - Adjusted styles for autofill, focus states, and padding in input fields to improve user experience.

- **Documentation**
  - Updated TextInput component documentation to reflect new `prefix` and `suffix` props usage and examples.

- **Refactor**
  - Removed `startIcon` and `endIcon` properties from TextInputProps and updated rendering logic.
  - Replaced `startIcon` with `prefix` in CurrencyInputComponent and PhoneInputComponent.
  - Changed `startIcon` and `endIcon` to `prefix` and `suffix` in InputComponent.

- **Style**
  - Removed specific styling for `[data-field-input-group]` affecting display and alignment.
  - Updated text-input module CSS to enhance the appearance of prefix and suffix elements and small-sized input fields.

- **Chores**
  - Added `MULTI_LINE_TEXT` constant to the `InputTypes` object.
  - Updated the label for the "Icon" property in WDSInputWidget's propertyPaneStyleConfig.

### CODE REVIEW

File: app/client/packages/design-system/headless/src/components/TextInput/src/types.ts [hunk 1]
```diff
@@ -19,10 +19,6 @@ export interface TextInputProps extends OmitedSpectrumTextFieldProps {
   fieldClassName?: string;
   /** className for the text input. */
   className?: string;
-  /**an icon to be displayed at the start of the component  */
-  startIcon?: React.ReactNode;
-  /** an icon to be displayed at the end of the component */
```
Comment (KelvinOm) [inline app/client/packages/design-system/headless/src/components/TextInput/src/types.ts:24]: Deleted `startIcon` and `endIcon` we'll use prefix and suffix instead.

File: app/client/packages/design-system/widgets/src/components/TextInput/src/TextInput.tsx [hunk 2]
```diff
@@ -30,14 +28,13 @@ const _TextInput = (props: TextInputProps, ref: HeadlessTextInputRef) => {
   const {
     contextualHelp: contextualHelpProp,
     description,
-    endIcon,
     errorMessage,
     isLoading = false,
     isRequired,
     label,
-    loaderPosition = "auto",
```
Comment (KelvinOm) [inline app/client/packages/design-system/widgets/src/components/TextInput/src/TextInput.tsx:38]: Deleted loaderPosition we'll use suffix as a slot.

File: app/client/packages/design-system/widgets/src/styles/src/text-input.module.css [hunk 3]
```diff
@@ -21,16 +20,22 @@
...
+    );
+    padding-block: var(--inner-spacing-3);
 
     &:autofill,
     &:autofill:hover,
     &:autofill:focus,
     &:autofill:active {
       font-size: initial;
+      -webkit-text-fill-color: var(--color-fg);
+      -webkit-box-shadow: 0 0 0 40rem var(--color-bg-neutral-subtle) inset;
```
Comment (jsartisan) [inline app/client/packages/design-system/widgets/src/styles/src/text-input.module.css:38]: What does this do?

**Review by jsartisan — DISMISSED** [review]

File: app/client/packages/design-system/widgets/src/styles/src/text-input.module.css (hunk 3 above)
Comment (KelvinOm) [reply app/client/packages/design-system/widgets/src/styles/src/text-input.module.css:39]: Left comment.

**Pull request merged**